### ISSUE TITLE
Why no waitForSync option on db.query?

### ISSUE THREAD

**Issue by tslocke** [arangodb/arangodb#16051]

**Comment by pluma** [comment]
The arangojs answer is that the HTTP API does not include that option: https://www.arangodb.com/docs/stable/http/aql-query-cursor-accessing-cursors.html

I've transferred this issue to the arangodb repo as that's the more appropriate place to question/discuss why that is the case.

**Comment by mpoeter** [comment]
The `waitForSync` option is not part of the query request itself, but it can be specified for each modification operation inside the query. E.g., for UPDATE see https://www.arangodb.com/docs/stable/aql/operations-update.html#waitforsync

That said, `waitForSync` will not help avoid write-write conflicts. `waitForSync` simply means that you want to wait until the modification operation has been flushed to disc before the response is sent back, but that won't prevent multiple concurrent updates to the same documents. If you need this, then you should consider using the `exclusive` flag, which can also be set for each modification operation individually inside the query, just like the `waitForSync`: https://www.arangodb.com/docs/stable/aql/operations-update.html#exclusive

**Comment by suneermehmood** [comment]
@mpoeter - looks like the Document API and py-arango library do not support `exclusive` flag ? Does it have to be an AQL query to use this flag ?

**Comment by aMahanna** [comment]
> @mpoeter - looks like the Document API and py-arango library do not support `exclusive` flag ? Does it have to be an AQL query to use this flag ?

@suneermehmood you are correct that the Document Update API (`patch /_api/document/{collection}/{key}`) does not support the `exclusive` flag, subsequently resulting in no `exclusive` support for the [python-arango update](https://docs.python-arango.com/en/main/specs.html#arango.collection.StandardCollection.update) method.

The only mention of `exclusive` in the python driver is for the [`beging_transaction`](https://docs.python-arango.com/en/main/specs.html#arango.database.StandardDatabase.begin_transaction) method, which seems unrelated to this matter.

P.S I am assuming that you are talking about [python-arango](https://github.com/ArangoDB-Community/python-arango), and not [pyArango](https://github.com/ArangoDB-Community/pyArango), as the latter is not officially maintained

**Comment by suneermehmood** [comment]
Yes, it is python-arango that I meant. Thank you for the confirmation @aMahanna
